glean-core/src/internal_metrics.rs

```diff
@@ -56,6 +63,15 @@ impl CoreMetrics {
                 disabled: false,
                 dynamic_label: None,
             }),
+
+            io_errors: CounterMetric::new(CommonMetricData {
+                name: "io".into(),
+                category: "glean.error".into(),
+                send_in_pings: vec!["metrics".into()],
```

Comment:
In the bug report you said:

> Maybe in the "baseline" ping? We want this failure to be observable since it might result in data loss.

For now I have it in the metrics ping. I'm not really sure where to put it though.